### ISSUE TITLE
Naming/FileName with rails template variants, Regex option

### ISSUE THREAD

**Issue by jaredmoody** [bbatsov/rubocop#4939]
I'm using `haml-lint` to run some rubocop checks on my view templates in my rails app.

Rubocop is complaining on variant view template file names that they're not using snake_case - e.g. on a file named `_product.html+tablet.haml`

1. Since this is Rails convention, maybe a good idea to include `+` in the Regex?
2. I attempted to just override this in my config:

```
Naming/FileName:
  Regex: /^[\+\da-z_.?!]+$/
```

Which gave me an error:  `type mismatch: String given`

Is there a different way I'm supposed to be defining the Regex option or is this a bug where the cop should be parsing the option as a regex and is not?

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.5 x86_64-darwin16)
```

**Comment by Drenmi** [comment]
You need to use the YAML syntax for regular expressions for it to be parsed as that. 🙂 

```yaml
Naming/FileName:
  Regex: !ruby/regexp '/^[\+\da-z_.?!]+$/'
```

**Comment by jaredmoody** [comment]
Aha, thanks!
